#1 [internal] load build definition from Dockerfile
#1 sha256:b5f6f28bc62fb39b3aec8704a76e3c9ca47da515f60fbf49fb1ef2a59c9c192f
#1 transferring dockerfile: 724B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:3706a9daab2037a343dfc91739a7f291a6b1d84fb3f4b1e3a6e09bcce94e1b2c
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/ubuntu:20.04
#3 sha256:cd9ed232facd22264e9246054f9c3d7a9c42bdb414a225d8f8215114646e640c
#3 DONE 0.2s

#4 [ 1/18] FROM docker.io/library/ubuntu:20.04@sha256:ed4a42283d9943135ed87d4ee34e542f7f5ad9ecf2f244870e23122f703f91c2
#4 sha256:8f291decb28dc3b4198102409bcdae24147c130a31cf61bcaa5d4e4fd9a825c1
#4 DONE 0.0s

#16 [internal] load build context
#16 sha256:45e9b8e7ec6702c812908cb1923fb168e8f48e65c354bbb88e945b5facce0bf3
#16 transferring context: 1.46MB 0.4s done
#16 DONE 0.4s

#5 [ 2/18] RUN apt-get update
#5 sha256:a7a70b9babceb8aabe1e5debe7f6add7f37aaa1213c5441c6c0085a626201fca
#5 CACHED

#10 [ 7/18] RUN apt-get install mysql-client --yes
#10 sha256:2745679caadab19ab15193f5f34860567a42ca824388289d165fb5fc9b2c9909
#10 CACHED

#15 [12/18] RUN python -m pip install direct_redis
#15 sha256:4a9224ed27dc2ccdcd002d83f5e5acbbb7f3b38d0640a6b30edc09206f230deb
#15 CACHED

#8 [ 5/18] RUN python -m pip uninstall -y bson
#8 sha256:757b36d72104bf3c38f7156b7e3af083e23ceb9300ff494868a5dccb37c183ce
#8 CACHED

#6 [ 3/18] RUN apt-get upgrade --yes
#6 sha256:779c22f0b850b34ff5680c564ea788ae6c91389b447d4b5f1bb822a13242cf75
#6 CACHED

#12 [ 9/18] RUN python -m pip install redis
#12 sha256:bc710fd748e3bf8b268f59d31cf9823bbf3042b3e870a0962542d19f1b412602
#12 CACHED

#14 [11/18] RUN python -m pip install cryptography
#14 sha256:499bf538b6dd15cdaf9257d92c9c8678ab35cdce53f1d6327420f034461b563b
#14 CACHED

#11 [ 8/18] RUN python -m pip install pymysql
#11 sha256:24a4f922451f7d6c267941842b4e28eb6084198928f20bb397974c4b54d0eb29
#11 CACHED

#7 [ 4/18] RUN apt-get install python3.8 python-is-python3 python3-pip --yes
#7 sha256:38b9e6c0906fbfea62cda624c2ae04b89e78e154eb0ee8960d192c8d1e0739bc
#7 CACHED

#17 [13/18] COPY ./requirements.txt /dbknitter/requirements.txt
#17 sha256:b16b4dd144fc88eeb16012da92a55d1ad2547a4a3199ae92fe767483fb190dc6
#17 CACHED

#13 [10/18] RUN python -m pip install cdifflib
#13 sha256:75cb4463443188da464d967d6712f9f6c294706c91b4e8e8e4a6dac0476097a2
#13 CACHED

#9 [ 6/18] RUN python -m pip install pandas pymongo
#9 sha256:3ef7d5d41d632169da7adbfd9008846d00fb40c20e9b9a8e6ab294158ef4aff6
#9 CACHED

#18 [14/18] RUN pip install -r /dbknitter/requirements.txt
#18 sha256:1812015db720cf079a775029797173df4fa0f4befc0e9c58ee16768e4fd92530
#18 CACHED

#19 [15/18] COPY . /dbknitter
#19 sha256:5a0a9d9f3276c8b7c4ad46606bd588e350ec724d242fe901fdf0362b977020f1
#19 DONE 1.3s

#20 [16/18] WORKDIR /dbknitter
#20 sha256:c4ff925a657299361a9c42e095e73019551f28eee4247d2b7b733448f28826e6
#20 DONE 0.0s

#21 [17/18] RUN pip install .
#21 sha256:8f9b1f3b2abc205827f364d71accf6b1aebc63d58258d88d900b3816e5f0d922
#21 0.748 Processing /dbknitter
#21 2.936 Requirement already satisfied: openai in /usr/local/lib/python3.8/dist-packages (from dbknitter==0.1.0) (1.3.5)
#21 2.943 Requirement already satisfied: requests in /usr/local/lib/python3.8/dist-packages (from dbknitter==0.1.0) (2.31.0)
#21 2.950 Requirement already satisfied: tqdm>4 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (4.66.1)
#21 2.957 Requirement already satisfied: anyio<4,>=3.5.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (3.7.1)
#21 2.982 Requirement already satisfied: pydantic<3,>=1.9.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (2.5.2)
#21 2.986 Requirement already satisfied: typing-extensions<5,>=4.5 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (4.8.0)
#21 2.987 Requirement already satisfied: httpx<1,>=0.23.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (0.25.2)
#21 2.993 Requirement already satisfied: distro<2,>=1.7.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (1.8.0)
#21 2.994 Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (2023.11.17)
#21 2.995 Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (3.6)
#21 2.996 Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (3.3.2)
#21 2.998 Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (2.1.0)
#21 3.001 Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.8/dist-packages (from anyio<4,>=3.5.0->openai->dbknitter==0.1.0) (1.3.0)
#21 3.002 Requirement already satisfied: exceptiongroup; python_version < "3.11" in /usr/local/lib/python3.8/dist-packages (from anyio<4,>=3.5.0->openai->dbknitter==0.1.0) (1.2.0)
#21 3.005 Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/lib/python3.8/dist-packages (from pydantic<3,>=1.9.0->openai->dbknitter==0.1.0) (0.6.0)
#21 3.008 Requirement already satisfied: pydantic-core==2.14.5 in /usr/local/lib/python3.8/dist-packages (from pydantic<3,>=1.9.0->openai->dbknitter==0.1.0) (2.14.5)
#21 3.010 Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.8/dist-packages (from httpx<1,>=0.23.0->openai->dbknitter==0.1.0) (1.0.2)
#21 3.016 Requirement already satisfied: h11<0.15,>=0.13 in /usr/local/lib/python3.8/dist-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai->dbknitter==0.1.0) (0.14.0)
#21 3.019 Building wheels for collected packages: dbknitter
#21 3.020   Building wheel for dbknitter (setup.py): started
#21 3.175   Building wheel for dbknitter (setup.py): finished with status 'done'
#21 3.175   Created wheel for dbknitter: filename=dbknitter-0.1.0-py3-none-any.whl size=30211 sha256=682c95d6c54d1fd74ae0aaaf0fd0d2d72a4d4898d5a5c9722494673f26cb7502
#21 3.175   Stored in directory: /tmp/pip-ephem-wheel-cache-2aez6_he/wheels/57/d4/17/cb4286ba457866ca85744b2bfc3c96e280880f7d32d199b163
#21 3.176 Successfully built dbknitter
#21 3.240 Installing collected packages: dbknitter
#21 3.261 Successfully installed dbknitter-0.1.0
#21 DONE 3.8s

#22 exporting to image
#22 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#22 exporting layers
#22 exporting layers 1.8s done
#22 writing image sha256:ce08bd5257d874ad4542b65552157aafd3bf2aec92780a90ba938f3f43c4368b done
#22 naming to docker.io/library/client-image done
#22 DONE 1.8s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:81efc28d7db82e54c49d00f277132402efe68e8c5ef2a583a8d4b44ea909e3fb
#1 transferring dockerfile: 541B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:54967b1ec92303b00793a5a18ada03f9a541500d38d3a0576003c81161e50689
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/mysql:latest
#3 sha256:581f9497edc116620469bb65354591ecdce3dfcf818d078c5f13986cd11b3f12
#3 DONE 0.2s

#4 [1/1] FROM docker.io/library/mysql:latest@sha256:1773f3c7aa9522f0014d0ad2bbdaf597ea3b1643c64c8ccc2123c64afd8b82b1
#4 sha256:c295488cd3d66d4c67c58595f61de0477f2d5456131b15e001ad4fa51c315311
#4 CACHED

#5 exporting to image
#5 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#5 exporting layers done
#5 writing image sha256:42cffdc4fb0382b53e3176e27efdec78cc28b8f172709bf41d208cf285395c72 done
#5 naming to docker.io/library/mysql-image done
#5 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:ad2bd6d83b7c5237add6177c863b5ed661c691ce1595fec6413f56b3a34d2f9e
#1 transferring dockerfile: 711B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:ecc57c2db237eb6dd151f4fecfc02ca41004300b37af893c3bd6c8e6df470742
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/mongodb/mongodb-community-server:6.0-ubi8
#3 sha256:db9fa2d3c04f6f47b7028de5a03cfa56ceb5b573185a7994e61708c70c2a55c7
#3 DONE 0.2s

#5 [1/2] FROM docker.io/mongodb/mongodb-community-server:6.0-ubi8@sha256:6b2289a866ad467299de456f872b2e6a77948c835377d2a61dce962e511b64d2
#5 sha256:a3309e74794f9c6c11f4e4dab176c6f626a458dc608a986a5aa1e0ec78484595
#5 DONE 0.0s

#4 [2/2] RUN python3 -m pip install pandas pymongo
#4 sha256:4c262285873f341ae337b4f4c6f3e906d35d1e6f169aa5d3bc69ad5d6ede9c96
#4 CACHED

#6 exporting to image
#6 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#6 exporting layers done
#6 writing image sha256:d4e7f20a2450efb1e761eaa46e5565825bf54d815b2cf9addedfef5e6faf2f91 done
#6 naming to docker.io/library/mongodb-image done
#6 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:87d37901e76c32dded86cdbb5363e5640e6b07e1228828cc3058896f0b43d854
#1 transferring dockerfile: 339B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:4a6332772efe176318e3f486fd1eb318ceb04172a6ee28d27349597236f363a4
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/redis:7.2.2-bookworm
#3 sha256:76d21ddd67c3b5083cfa98c46d64e994eb755bbb2cabe0b45bf1ab438b3ea244
#3 DONE 0.2s

#4 [1/6] FROM docker.io/library/redis:7.2.2-bookworm@sha256:4b0069ac29dd0ee5d3aec7ee34fa1034017a4c3a2c094751a2412cc609e6dea3
#4 sha256:2b3d8b9e7dd3596835f96d0a826fa018ce98773f6a60c06055ec101a53cc944c
#4 DONE 0.0s

#5 [2/6] RUN apt update && apt -y upgrade
#5 sha256:b42a3e0cb7cac3cde566cd0c1088380f6ea2a8a91bef0ab724b067a25a85cd3f
#5 CACHED

#6 [3/6] RUN apt install -y python3 python3-pip
#6 sha256:e143120aff70fc59ecc5517e880a35a29862aece39eeb2f2b23b5faee9b14656
#6 CACHED

#7 [4/6] RUN apt install -y build-essential libssl-dev libffi-dev python3-dev
#7 sha256:b8f6b1866594989e265224e447ea352f251e6023b78aacec8cf8c9b5aaa0fde2
#7 CACHED

#8 [5/6] RUN apt install -y python3.11-venv
#8 sha256:248ec744bb080b2c8cbce469ab62c2fd3d0b545262a232f0fec9c66190b06a5f
#8 CACHED

#9 [6/6] RUN python3 -m venv /venv && . /venv/bin/activate && pip install pandas redis direct_redis
#9 sha256:efe492789e071b3d7bcfced230528ea90fe616a058a4e9fa0d03f9b875c89c62
#9 CACHED

#10 exporting to image
#10 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#10 exporting layers done
#10 writing image sha256:eea1c4945676af93e6f5192e56e7e6f78f1ac8d0716d8d8244d0b8423fa57509 done
#10 naming to docker.io/library/redis-image done
#10 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Network cloudlab_default  Creating
Network cloudlab_default  Created
Container cloudlab-mysql-1  Creating
Container cloudlab-client-1  Creating
Container cloudlab-mongodb-1  Creating
Container cloudlab-redis-1  Creating
Container cloudlab-client-1  Created
Container cloudlab-redis-1  Created
Container cloudlab-mysql-1  Created
Container cloudlab-mongodb-1  Created
